File: solutions/javascript/sieve/1/sieve.js
export const primes = (maxNumber) => {
  // Если число меньше 2, простых чисел нет
  if (maxNumber < 2) return [];

  // 1. Создаем список, где сначала все числа считаем простыми (true)
  const isPrime = new Array(maxNumber + 1).fill(true);
  
  // Числа 0 и 1 — не простые
  isPrime[0] = false;
  isPrime[1] = false;

  // 2. Начинаем просеивать числа
  for (let i = 2; i * i <= maxNumber; i++) {
    // Если число i отмечено как простое
    if (isPrime[i]) {
      // Вычеркиваем все числа, которые делятся на i
      for (let j = i * i; j <= maxNumber; j += i) {
        isPrime[j] = false;
      }
    }
  }

  // 3. Собираем все оставшиеся простые числа в один список
  const result = [];
  for (let i = 2; i <= maxNumber; i++) {
    if (isPrime[i]) {
      result.push(i);
    }
  }

  return result;
};

File: solutions/julia/sieve/1/sieve.jl
function sieve(limit)
 
    if limit < 2
        return Int[]
    end

  
    is_prime = fill(true, limit)
    is_prime[1] = false # 1 — не простое число


    for i in 2:isqrt(limit)
        if is_prime[i]
       
            for j in (i^2):i:limit
                is_prime[j] = false
            end
        end
    end


    return [i for i in 1:limit if is_prime[i]]
end
